plugin-vidarr/src/main/java/ca/on/oicr/gsi/shesmu/vidarr/VidarrPlugin.java

```diff
@@ -50,6 +54,107 @@
 
 public class VidarrPlugin extends JsonPluginFile<Configuration> {
 
+  private class AnalysisCache extends ValueCache<Stream<VidarrAnalysisValue>> {
+    public AnalysisCache(Path fileName) {
+      super("vidarr-analysis " + fileName.toString(), 30, ReplacingRecord::new);
+    }
+
+    private String createVidarrProvenanceRequestBody(
+        List<String> analysisTypes, List<String> versionTypes) {
+      return "{"
+          + "\"analysisTypes\": ["
+          + analysisTypes.stream()
+              .map(name -> ("\"" + name + "\""))
+              .collect(Collectors.joining(","))
+          + "],"
+          + "\"epoch\": 0,"
+          + "\"includeParameters\": false,"
+          + "\"timestamp\": 0,"
+          + "\"versionPolicy\": \"LATEST\","
+          + "\"versionTypes\": ["
+          + versionTypes.stream().map(name -> ("\"" + name + "\"")).collect(Collectors.joining(","))
+          + "]}";
+    }
+
+    protected Stream<VidarrAnalysisValue> analysisArchive(String baseUrl) throws Exception {
+
+      final var results =
+          HTTP_CLIENT.send(
+              HttpRequest.newBuilder(URI.create(baseUrl + "/api/provenance"))
+                  .header("Content-type", "application/json")
+                  .timeout(Duration.ofMinutes(10))
+                  .POST(
+                      HttpRequest.BodyPublishers.ofString(
+                          createVidarrProvenanceRequestBody(
+                              configuration.get().getAnalysisTypes(),
+                              configuration.get().getVersionTypes())))
+                  .build(),
+              new JsonBodyHandler<>(
+                  MAPPER, new TypeReference<AnalysisProvenanceResponse<ExternalKey>>() {}));
+
+      if (results.statusCode() != 200) {
+        return new ErrorableStream<>(Stream.empty(), false);
+      }
+
+      try {
+        final var body = results.body().get();
+        return body.getResults().stream()
+            .map(
+                ca ->
+                    new VidarrAnalysisValue(
+                        ca.getCompleted() == null
+                            ? Optional.empty()
+                            : Optional.of(ca.getCompleted().toInstant()),
+                        ca.getAnalysis().stream()
```

Comment:
If I make analysisTypes an empty array, this line attempts to call stream() on null. It doesn't really make sense to make analysisTypes empty, but there should be an earlier and clearer error in that case

Reply:
this is also true of setting analysisTypes to just 'URL' enough though that should be a valid configuration (just with no records)